src/Main.elm

```diff
@@ -1,10 +1,251 @@
+      }
+    , Cmd.none
+    )
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        Select id ->
+            ( { model | currentFolderId = Just id }, Cmd.none )
+
+        GoUp id ->
+            ( { model | currentFolderId = id }, Cmd.none )
+
+        Filter type_ ->
+            ( { model | filter = type_, currentFolderId = Nothing }, Cmd.none )
+
+
+view : Model -> Html Msg
+view model =
+    div []
+        [ -- display the buttons to filter by type_,
+          List.map
+            (\type_ ->
+                button
+                    [ onClick (Filter type_)
+                    , class "button-group-button"
+                    , if model.filter == type_ then
+                        style "opacity" "0.5"
+
+                      else
+                        style "opacity" "1"
+                    ]
+                    [ text (Debug.toString type_) ]
+            )
+            allTypes
+            |> div [ class "button-group" ]
+
+        --folders are handled here
+        , List.map
+            (\folder ->
+                let
+                    --is the current folder opened? (only one folder can be opened at a time)
+                    opened =
+                        case model.currentFolderId of
+                            Just id ->
+                                if folder.id == id then
+                                    True
+
+                                else
+                                    False
+
+                            Nothing ->
+                                False
+                in
+                div
+                    []
+                    [ Data.AudienceFolder.view folder opened (Select folder.id) (GoUp folder.parent)
+                    ]
+            )
+            (case model.filter of
+                Data.Audience.Shared ->
+                    []
+
+                _ ->
+                    filterAudienceFolders model.folders model.currentFolderId
+            )
+            |> div []
+
+        --audiences are handled here
+        , div []
+            [ List.map
+                (\audience ->
+                    Data.Audience.view audience
+                )
+                (case model.filter of
+                    Data.Audience.Shared ->
+                        filterAudienceCategory model.audiences Data.Audience.Shared
+
+                    filterValue ->
+                        filterAudiences (filterAudienceCategory model.audiences filterValue) model.currentFolderId
+                )
+                |> div []
+            ]
+        ]
+
+
+filterAudienceCategory : List Audience -> Data.Audience.AudienceType -> List Audience
+filterAudienceCategory audiences type_ =
+    {- | Helper function to use with List.filter, separate audiences by type_ -}
+    List.filter
+        (\e ->
+            if e.type_ == type_ then
+                True
+
+            else
+                False
+        )
```

Comment:
```suggestion
        (\e ->  e.type_ == type_)
```

Reply:
No idea how I didn't realize this sooner, so thank you for opening my eyes 😉 Now I can get rid of all those verbose if elses...